python/tank/context.py

```diff
@@ -984,7 +1107,36 @@ def context_yaml_constructor(loader, node):
 ################################################################################################
 # utility methods
 
-def _task_from_sg(tk, task_id):
+def _get_entity_type_sg_name_field(entity_type):
```

Comment:
I centralised it for this file but based on past experience with name in the path cache, I'm reluctant to make bigger changes to code outside of this scope that will potentially lead down a rabbit hole!  I'll leave it as is for now but will create a separate ticket to rationalise/centralise this logic :)